src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/hosts/hosts.component.ts

```diff
@@ -70,46 +84,31 @@ export class HostsComponent extends ListWithDetails implements OnInit {
         name: this.actionLabels.CREATE,
         permission: 'create',
         icon: Icons.add,
-        click: () => {
-          this.depCheckerService.checkOrchestratorOrModal(
-            this.actionLabels.CREATE,
-            $localize`Host`,
-            () => {
-              this.router.navigate([this.urlBuilder.getCreate()]);
-            }
-          );
-        }
+        click: () => this.router.navigate([this.urlBuilder.getCreate()]),
+        disable: (selection: CdTableSelection) => Boolean(this.getDisableDesc('create', selection)),
+        disableDesc: (selection: CdTableSelection) => this.getDisableDesc('create', selection)
       },
       {
         name: this.actionLabels.EDIT,
         permission: 'update',
         icon: Icons.edit,
-        click: () => {
-          this.depCheckerService.checkOrchestratorOrModal(
-            this.actionLabels.EDIT,
-            $localize`Host`,
-            () => this.editAction()
-          );
-        },
+        click: () => this.editAction(),
         disable: (selection: CdTableSelection) =>
-          !selection.hasSingleSelection || !selection.first().sources.orchestrator,
-        disableDesc: this.getEditDisableDesc.bind(this)
+          !selection.hasSingleSelection ||
+          !selection.first().sources.orchestrator ||
+          Boolean(this.getDisableDesc('edit', selection)),
```

Comment:
> Both validations are already done in `getDisableDesc`.
> same in other places...

`getDisableDesc` is not suitable for disable state because in some cases, the button is disabled and there is no description text for it. e.g. Select nothing and hover to the `Delete` or `Edit` button.

Reply:
I made `getDisable` return an object:
```typescript
  getDisable(
    action: 'create' | 'edit' | 'delete',
    selection: CdTableSelection
  ): { state: boolean; desc?: string } {
```

And use the corresponding property:

```typescript
disable: (selection: CdTableSelection) => this.getDisable('create', selection).state,
disableDesc: (selection: CdTableSelection) => this.getDisable('create', selection).desc
```